### DISCUSSION TITLE
Is it possible to set custom traceId ? To have possibility to search traces&spans via that custom traceId

### DISCUSSION THREAD

**Discussion by VladimirTitov4** [jaegertracing/jaeger#3618] [General]
Hello everyone. Can't find how to configure jaeger to use custom traceId. 
I'm using spring boot.

**Comment by frzifus** [comment]
As far as i know, there is nothing like this. Also, i'm not sure if this would make much sense, since you want to distinguish traces based on individual ids. But depending on what exactly you have in mind, there is a way to propagate [key/value](https://github.com/opentracing/specification/blob/master/specification.md#set-a-baggage-item) pairs. May that helps?

**Reply by VladimirTitov4** [reply]
Thank you for your answer. Actually I need to search traces for some **MY_CUSTOM_KEY** 

Actually I need to analyze requests in retrospective. So I need to store somewhere all the traces, e.g. for a week, store traceId and then search with that traceId what went wrong. 

If there is no way to use custom traceId, I could use a Map with key as my custom id and associate with jaeger produced traceId. 

---
You mentioned that there is a way to propagate with key/value pairs. But how to search specific trace with that key/value pair ? In jaeger UI I see only option to search with traceId

**Reply by frzifus** [reply]
Maybe there is a way to set Baggage-Items as Span-Tags automatically. 
With the use of such tags, it would be possible to filter them. e.g.: `custom.Key=1234-567`

![image](https://user-images.githubusercontent.com/10403402/162757142-eef0613d-263a-4a67-9ee3-d976b88efbc2.png)

But if I understood the whole thing correctly, it would also be enough to set a unique tag at any point. Via the tag the spanID can be found out and via the SpanID then the traceID.

Did i miss something @pavolloffay ?

**Reply by VladimirTitov4** [reply]
> Thank you for your answer. Actually I need to search traces for some **MY_CUSTOM_KEY**
> 
> Actually I need to analyze requests in retrospective. So I need to store somewhere all the traces, e.g. for a week, store traceId and then search with that traceId what went wrong.
> 
> If there is no way to use custom traceId, I could use a Map with key as my custom id and associate with jaeger produced traceId.
> 
> You mentioned that there is a way to propagate with key/value pairs. But how to search specific trace with that key/value pair ? In jaeger UI I see only option to search with traceId

Actually I found a solution, I just set tag to a span. 
with having a bean injected ``` private final JaegerTracer tracer; ```
and adding new tag
```
Span span = tracer.activeSpan();
span.setTag(SENTRY_TRACE_ID, "MY_CUSTOM_TRACE_ID");
```

but how to add specific tag to a whole trace, but not a specific span ?

**Reply by VladimirTitov4** [reply]
> 

Thank you for your answer!
Actually in have 2 questions in additions

1. How to set tags automatically ? 
or
2. How to set a tag for a whole trace, but not for specific span ?

**Reply by yurishkuro** [reply]
1 - there is no way. Depending on the instrumentation you're using there might be callbacks you could hook into, but ultimately you are setting a _custom tag_, so by definition it cannot be set _automatically_.
2 - there's no storage entity for a whole trace, only spans. By convention, trace-level attributes are set on the root span (e.g. sampling parameters).

**Reply by VladimirTitov4** [reply]
Thank you for your answer. 

Actually I managed to add my traceId as a tag via SpringBootRequestInterceptor

```
@Service
@RequiredArgsConstructor
public class TracingInterceptor extends OncePerRequestFilter {

    private final JaegerTracer tracer;

    @Override
    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
        String sentryHeader = request.getHeader(SENTRY_TRACE_ID);
        Span span = tracer.activeSpan();
        if (sentryHeader != null && span != null) {
            String[] split = sentryHeader.split(":");
            String sentryTraceId = split[0];
            span.setTag(SENTRY_TRACE_ID, sentryTraceId);
        }
        filterChain.doFilter(request, response);
    }
}
```
